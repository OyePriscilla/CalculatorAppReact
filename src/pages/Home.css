.divClass {
  width: 90%;
  margin: 0 auto;
}

h2 {
  text-align: left;
  color: #707075;
}

p {
  font-size: 20px;
  margin-top: 50px;
  text-align: justify;
}
