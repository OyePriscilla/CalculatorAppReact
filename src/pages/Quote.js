import React from 'react';

export default function Quote() {
  return (
    <div>
      <p style={{ margin: '230px 40px', fontWeight: '600' }}>
        Mathematics is not about numbers, equations, computations, or algorithms:
        it is about understanding. -William Paul Thurston
      </p>
    </div>
  );
}
