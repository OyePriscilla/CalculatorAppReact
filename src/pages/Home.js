import React from 'react';
import './Home.css';

export default function Home() {
  return (
    <div className="divClass">
      <h2>Welcome to Calculator App Page!</h2>
      <p>
        This calculator can be used to do basic arithemetic operations like
        addition, subtraction, division, multiplication e.t.c. Do your calculation anywhere
        and anytime of the day! Feel free to use it.
      </p>
    </div>
  );
}
