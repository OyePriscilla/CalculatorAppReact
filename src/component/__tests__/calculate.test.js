import renderer from 'react-test-renderer';
import Calculator from '../calculator';
import Home from '../../pages/Home';
import Quote from '../../pages/Quote';

describe('App component testing with snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Home />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Quote />);
    expect(tree).toMatchSnapshot();
  });
});
