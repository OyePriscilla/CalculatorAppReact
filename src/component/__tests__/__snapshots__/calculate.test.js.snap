// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component testing with snapshot renders correctly 1`] = `
<div
  data-testid="calc"
  style={
    Object {
      "display": "flex",
      "justifyContent": "spaceEvenly",
      "margin": "50px",
    }
  }
>
  <div
    style={
      Object {
        "fontSize": "24px",
        "fontWeight": "900",
        "marginLeft": "120px",
      }
    }
  >
    Let's do some math!
  </div>
  <div
    className="container"
  >
    <div
      className="top"
      onChange={[Function]}
    >
      0
    </div>
    <div
      className="wrapper"
    >
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="no-zero-class operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="no-zero-class operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="no-zero-class operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="no-zero-class operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero-class"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="no-zero-class"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="no-zero-class operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`App component testing with snapshot renders correctly 2`] = `
<div
  className="divClass"
>
  <h2>
    Welcome to Calculator App Page!
  </h2>
  <p>
    This calculator can be used to do basic arithemetic operations like addition, subtraction, division, multiplication e.t.c. Do your calculation anywhere and anytime of the day! Feel free to use it.
  </p>
</div>
`;

exports[`App component testing with snapshot renders correctly 3`] = `
<div>
  <p
    style={
      Object {
        "fontWeight": "600",
        "margin": "230px 40px",
      }
    }
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston
  </p>
</div>
`;
